daemon/containerd/image_exporter.go

```diff
@@ -68,19 +68,18 @@ func (i *ImageService) ExportImage(ctx context.Context, names []string, platform
 		archive.WithSkipMissing(i.content),
 	}
 
-	leasesManager := i.client.LeasesService()
-	lease, err := leasesManager.Create(ctx, leases.WithRandomID())
+	ctx, done, err := i.client.WithLease(ctx)
 	if err != nil {
 		return errdefs.System(err)
 	}
 	defer func() {
-		if err := leasesManager.Delete(ctx, lease); err != nil {
+		if err := done(context.WithoutCancel(ctx)); err != nil {
```

Comment:
In the export case there really isn't a need to hold onto anything since even if the operation is canceled, it will need to restart from the beginning on retry. For the pull case, a cancel may want to keep progress so the context error could be respected. We could make this more explicit and check for a context error rather than let the function error from it.